StripePaymentSheet/StripePaymentSheet/PaymentSheet/Intent.swift

```diff
@@ -174,20 +189,33 @@ extension STPConfirmPaymentMethodOptions {
         We read the top-level SFU to know the merchant’s desired save behavior
         We write payment method options SFU to set the customer’s desired save behavior
      */
-    func setSetupFutureUsageIfNecessary(_ shouldSave: Bool, paymentMethodType: STPPaymentMethodType) {
-        guard paymentMethodType == .card || paymentMethodType == .USBankAccount else {
-            // Only support card and US bank setup_future_usage in payment_method_options
+    func setSetupFutureUsageIfNecessary(
+        _ shouldSave: Bool,
+        paymentMethodType: STPPaymentMethodType,
+        customer: PaymentSheet.CustomerConfiguration?
+    ) {
+        // When no customer is set this property cannot be set.
+        // Only support card and US bank setup_future_usage in payment_method_options
+        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount
```

Comment:
I wonder if we should `assert(!(shouldSave && customer == nil))`

Reply:
Added